File: string/6.zig-zag-conversion.py
#
# @lc app=leetcode id=6 lang=python3
#
# [6] ZigZag Conversion
#

# @lc code=start


class Solution:
    def convert(self, s: str, numRows: int) -> str:
        if numRows == 1:
            return s

        rows = [''] * numRows

        col = 0
        direction = 1
        for i in range(len(s)):
            rows[col] += s[i]
            col += direction
            if col == 0 or col == numRows-1:
                direction *= -1

        ans = ""

        for j in range(numRows):
            ans += rows[j]

        return ans
# @lc code=end

File: Golang/6.zig-zag-conversion.go
/*
 * @lc app=leetcode id=6 lang=golang
 *
 * [6] ZigZag Conversion
 */

// @lc code=start
func convert(s string, numRows int) string {
	if numRows == 1 {
		return s
	}

	ans := make([]string, numRows)

	col, direction := 0, 1
	for i := 0; i < len(s); i++ {
		ans[col] += string(s[i])
		col += direction
		if col == 0 || col == numRows-1 {
			direction *= -1
		}
	}

	res := ""
	for i := 0; i < numRows; i++ {
		res += ans[i]
	}

	return res
}

// @lc code=end

